commit fd65418f600874b05f902b622078b40bc1abb24a
Author: Eric Anholt <eric@anholt.net>
Date:   Mon Aug 3 14:27:41 2009 -0700

    intel: Fix inverted test for disabling flushing of front buffer output.
    
    The comment disagreed with the code, and nicely drew my eyes to what was
    going wrong.
    
    Bug #21774 (blender)
    Bug #21788 (readpix)

diff --git a/src/mesa/drivers/dri/intel/intel_context.c b/src/mesa/drivers/dri/intel/intel_context.c
index 7f5b8d7..35d9985 100644
--- a/src/mesa/drivers/dri/intel/intel_context.c
+++ b/src/mesa/drivers/dri/intel/intel_context.c
@@ -513,7 +513,7 @@ intel_flush(GLcontext *ctx, GLboolean needs_mi_flush)
 	  * each of N places that do rendering.  This has worse performances,
 	  * but it is much easier to get correct.
 	  */
-	 if (intel->is_front_buffer_rendering) {
+	 if (!intel->is_front_buffer_rendering) {
 	    intel->front_buffer_dirty = GL_FALSE;
 	 }
       }
